From 21cde7fde306b3a4699fff415c90df457f5b30c7 Mon Sep 17 00:00:00 2001
From: Roland Mainz <rmainz@redhat.com>
Date: Thu, 9 Apr 2015 19:56:45 +0200
Subject: [PATCH] bug #140: Remove -fno-strict-aliasing

Remove -fno-strict-aliasing (this is not required because gssproxy
is mostly a good ISO C99/C11 citizen) and replace it with
-Werror=strict-aliasing to ensure that if *anything* creeps up
the build will just fail (this requires in gcc4.x's case the
use of -fstrict-aliasing, too).

References:
    - http://dbp-consulting.com/tutorials/StrictAliasing.html

Signed-off-by: Roland Mainz <rmainz@redhat.com>
---
 proxy/Makefile.am | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/proxy/Makefile.am b/proxy/Makefile.am
index 1e7c39cf2c2d3ef2c96c1b28c76d53d378054da6..6e38cf8116b70d51fe873720a2205572402d9f99 100644
--- a/proxy/Makefile.am
+++ b/proxy/Makefile.am
@@ -36,10 +36,12 @@ if WANT_AUX_INFO
     AM_CFLAGS += -aux-info $@.X
 endif
 if HAVE_GCC
+# -fstrict-aliasing is needed so that -W*strict-aliasing works
+# properly
     AM_CFLAGS += -Wall -Wshadow -Wstrict-prototypes -Wpointer-arith \
                  -Wcast-qual -Wcast-align -Wwrite-strings \
-                 -Werror-implicit-function-declaration \
-                 -fno-strict-aliasing
+                 -fstrict-aliasing -Wstrict-aliasing -Werror=strict-aliasing \
+                 -Werror-implicit-function-declaration
 endif
 
 dist_pkgconfig_DATA =
-- 
1.9.3

